src/corva/api_adapter.py

```diff
@@ -0,0 +1,207 @@
+        .strip("Headers()")  # strip obsolete data
+        .replace(
+            "'", '"'
+        )  # replace single quotes with double ones to get proper json string
+    )
+
+
+def _failed_request_msg(
+    msg: str,
+    request: httpx.Request,
+    response: Optional[httpx.Response],
+) -> str:
+    data = {"message": f"Request failed - {msg}"}
+
+    if response:
+        # log response first, so there is less chance it gets truncated
+        # and users are able to see server error message
+        data["response"] = {
+            "code": response.status_code,
+            "reason": response.reason_phrase,
+            "headers": _httpx_headers_to_dict(response.headers),
+            "content": str(response.content),
+        }
+
+    data["request"] = {
+        "method": request.method,
+        "url": str(request.url),
+        "headers": _httpx_headers_to_dict(request.headers),
+        "content": str(request.content),
+    }
+
+    # use yaml because it is much more readable in logs
+    return yaml.dump(data, sort_keys=False)
+
+
+def logging_send(func: Callable, *, logger: logging.Logger) -> Callable:
+    @functools.wraps(func)
+    def wrapper(request: httpx.Request, *args, **kwargs):
+        try:
+            response = func(request, *args, **kwargs)
+        except httpx.HTTPError as exc:
+            # Response was not received at all
+            logger.error(
+                _failed_request_msg(msg=str(exc), request=request, response=None)
+            )
+            raise
+
+        try:
+            response.raise_for_status()
+        except httpx.HTTPStatusError as exc:
+            # Response has unsuccessful status
+            logger.error(
+                _failed_request_msg(msg=str(exc), request=request, response=response)
+            )
+
+        return response
+
+    return wrapper
+
+
+class DataApiV1Sdk:
+    def __init__(self, client: httpx.Client):
+        self.http = client
+
+    def get(
+        self,
+        provider: str,
+        dataset: str,
+        *,
+        query: dict,
+        sort: dict,
+        limit: int,
+        skip: int = 0,
+        fields: Optional[str] = None,
+    ) -> List[dict]:
+        """Fetches data from the endpoint GET 'data/{provider}/{dataset}/'.
+
+        Args:
+            provider: company name owning the dataset.
+            dataset: dataset name.
+            query: search conditions. Example: {"asset_id": 123} - will fetch data
+                for asset with id 123.
+            sort: sort conditions. Example: {"timestamp": 1} - will sort data
+                in ascending order by timestamp.
+            limit: number of data points to fecth.
+                Recommendation for setting the limit:
+                    1. The bigger ↑ each data point is - the smaller ↓ the limit;
+                    2. The smaller ↓ each data point is - the bigger ↑ the limit.
+            skip: exclude from the response the first N items of the dataset.
```

Comment:
Can you add a comment that skip should be used only for small amount data? Having a large skip will lead to very slow queries.

Reply:
Done